platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentActivationOptions.java

```diff
@@ -0,0 +1,170 @@
+    this.context = context;
+    this.style = style;
+    this.locationEngine = locationEngine;
+    this.locationEngineRequest = locationEngineRequest;
+    this.locationComponentOptions = locationComponentOptions;
+    this.styleRes = styleRes;
+    this.useDefaultLocationEngine = useDefaultLocationEngine;
+  }
+
+  /**
+   * Convenience method to retrieve a {@link LocationComponentActivationOptions} object which is ready to build with
+   *
+   * @return a builder object
+   */
+  @NonNull
+  public static Builder builder(@NonNull Context context, @NonNull Style fullyLoadedMapStyle) {
+    return new LocationComponentActivationOptions.Builder(context, fullyLoadedMapStyle);
+  }
+
+  /**
+   * The application's current context
+   *
+   * @return the application's current context
+   */
+  @Nullable
+  public Context context() {
+    return context;
+  }
+
+  /**
+   * The proxy object for current map style. More info at {@link Style}
+   *
+   * @return the map's fully loaded Style object
+   */
+  @Nullable
+  public Style style() {
+    return style;
+  }
+
+  /**
+   * The {@link LocationEngine} which the {@link LocationComponent} should use
+   *
+   * @return the engine, or null if you'd like to only force location updates
+   */
+  @Nullable
+  public LocationEngine locationEngine() {
+    return locationEngine;
+  }
+
+  /**
+   * The location request which the {@link LocationComponent} should use
+   * @return
+   */
+  @Nullable
+  public LocationEngineRequest locationEngineRequest() {
+    return locationEngineRequest;
+  }
+
+  /**
+   * A built {@link LocationComponentOptions} object, which holds the various {@link LocationComponent} styling options
+   * @return the options for styling the actual LocationComponent
+   */
+  @Nullable
+  public LocationComponentOptions locationComponentOptions() {
+    return locationComponentOptions;
+  }
+
+  /**
+   * the LocationComponent style res
+   *
+   * @return
+   */
+  @Nullable
+  public int styleRes() {
+    return styleRes;
+  }
+
+  /**
+   * true if you want to initialize and use the built-in location engine or false if there should be no
+   * location engine initialized
+   * @return whether the default LocationEngine is used
+   */
+  @Nullable
+  public boolean useDefaultLocationEngine() {
+    return useDefaultLocationEngine;
+  }
+
+  /**
+   * Builder class for constructing a new instance of {@link LocationComponentActivationOptions}.
+   *
+   * since 7.2.0
+   */
+  public static class Builder {
+    private Context context;
+    private Style style;
+    private LocationEngine locationEngine;
+    private LocationEngineRequest locationEngineRequest;
+    private LocationComponentOptions locationComponentOptions;
+    private int styleRes;
+    private boolean useDefaultLocationEngine;
+
+    public Builder(Context context, Style style) {
+      this.context = context;
+      this.style = style;
+    }
+
+    public Builder() {
+
+    }
```

Comment:
Would propose to use one constructor: either go with one that showcases required fields or go with a no args constructor

Reply:
I'd go with the one that takes the required parameters and made the default one private.

Reply:
I don't even think the `no args` constructor is needed. I've removed for now. `Context` and `Style` will always be needed for now, so I've made that the only constructor.